dpreference/src/main/java/me/dozen/dpreference/PrefAccessor.java

```diff
@@ -1,94 +1,166 @@
-        Uri URI = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_BOOLEAN);
+        Uri uri = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_BOOLEAN);
         int value = defaultValue ? 1 : 0;
-        Cursor cursor = context.getContentResolver().query(URI, null, null, null, null);
-        if (cursor != null && cursor.moveToFirst()) {
-            value = cursor.getInt(cursor.getColumnIndex(PreferenceProvider.PREF_VALUE));
+        ResultSet cursor = null;
+        DataAbilityPredicates dataAbilityPredicates = new DataAbilityPredicates((java.lang.String) null);
+        dataAbilityPredicates.setWhereArgs(Arrays.asList());
+        dataAbilityPredicates.setOrder(null);
+        try {
+            cursor = DataAbilityHelper.creator(context).query(uri, null, dataAbilityPredicates);
+        } catch (DataAbilityRemoteException e) {
+            HiLog.error(HILOG_LABEL, "exception in getBoolean");
+        }
+        if (cursor != null && cursor.goToFirstRow()) {
+            value = cursor.getInt(cursor.getColumnIndexForName(PreferenceProvider.PREF_VALUE));
         }
         IOUtils.closeQuietly(cursor);
         return value == 1;
     }
 
     public static void remove(Context context, String name, String key) {
-        Uri URI = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_STRING);
-        context.getContentResolver().delete(URI, null, null);
+        Uri uri = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_STRING);
+        DataAbilityPredicates dataAbilityPredicates = new DataAbilityPredicates((java.lang.String) null);
+        dataAbilityPredicates.setWhereArgs(Arrays.asList());
+        try {
+            DataAbilityHelper.creator(context).delete(uri, dataAbilityPredicates);
+        } catch (DataAbilityRemoteException e) {
+            HiLog.error(HILOG_LABEL, "exception in remove");
+        }
     }
 
     public static void setString(Context context, String name, String key, String value) {
-        Uri URI = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_STRING);
-        ContentValues cv = new ContentValues();
-        cv.put(PreferenceProvider.PREF_KEY, key);
-        cv.put(PreferenceProvider.PREF_VALUE, value);
-        context.getContentResolver().update(URI, cv, null, null);
+        Uri uri = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_STRING);
+        ValuesBucket cv = new ValuesBucket();
+        cv.putString(PreferenceProvider.PREF_KEY, key);
+        cv.putString(PreferenceProvider.PREF_VALUE, value);
+        DataAbilityPredicates dataAbilityPredicates = new DataAbilityPredicates((java.lang.String) null);
+        dataAbilityPredicates.setWhereArgs(Arrays.asList());
+        try {
+            DataAbilityHelper.creator(context).update(uri, cv, dataAbilityPredicates);
+        } catch (DataAbilityRemoteException e) {
+            HiLog.error(HILOG_LABEL, "exception in getIntFromAttr");
```

Comment:
please use the method name in the log statement, to make it easier to identify during developer debug.

Reply:
Done